src/main/java/bisq/price/spot/providers/Yadio.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.price.spot.providers;
+
+import bisq.price.spot.ExchangeRate;
+import bisq.price.spot.ExchangeRateProvider;
+import bisq.price.util.yadio.YadioTicker;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.core.env.Environment;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.springframework.boot.context.properties.bind.Bindable.setOf;
+
+/**
+ * Yadio is used for "marginal market" currencies. Originally thought for:
+ * <ul>
+ *     <li>PYG (Paraguayan Guarani</li>
+ *     <li>DOP (Dominican Peso)</li>
+ *     <li>BOB (Bolivian Bolivariano</li>
+ *     <li>EGP (Egyptian Pound</li>
```

Comment:
Nit, you are missing closing parenthesis on a few of these currency names.
```suggestion
 *     <li>EGP (Egyptian Pound)</li>
```